File: deps/GraphBLAS/GraphBLAS/@GrB/private/util/gb_is_float.c
//------------------------------------------------------------------------------
// gb_is_float: check if a GrB_Type is floating-point
//------------------------------------------------------------------------------

// SuiteSparse:GraphBLAS, Timothy A. Davis, (c) 2017-2022, All Rights Reserved.
// SPDX-License-Identifier: GPL-3.0-or-later

//------------------------------------------------------------------------------

#include "gb_interface.h"

bool gb_is_float (const GrB_Type type)
{ 
    return ((type == GrB_FP32 ) ||
            (type == GrB_FP64 ) ||
            (type == GxB_FC32 ) ||
            (type == GxB_FC64 )) ;
}

File: deps/GraphBLAS/GraphBLAS/@GrB/private/gb_isfloat.m
function s = gb_isfloat (type)
%GB_ISFLOAT true for floating-point GraphBLAS types.
% Implements s = isfloat (type (G))

% SuiteSparse:GraphBLAS, Timothy A. Davis, (c) 2017-2022, All Rights Reserved.
% SPDX-License-Identifier: GPL-3.0-or-later

s = gb_contains (type, 'double') || gb_contains (type, 'single') ;

